metaverse_modules/heal-effect/index.js

```diff
@@ -0,0 +1,426 @@
+                  
+                
+                  if(vId>0.2 && vId<=1.)
+                      gl_FragColor = flash;
+                  else if(vId>=1.5)
+                      gl_FragColor = sparkle;
+                  else if(vId<0.2)
+                      gl_FragColor = circle;
+                  if(vId>=0. && vId<=1.)
+                      gl_FragColor.rgb *= vec3(0.0376, 0.940, 0.474);
+                  else
+                      gl_FragColor.rgb *= vec3(0.444, 0.999, 0.777);
+                  if(vId<0.2)
+                      gl_FragColor.a *= distance(avatarPos,vPos)*2.;
+                  ${THREE.ShaderChunk.logdepthbuf_fragment}
+              }
+          `,
+          side: THREE.DoubleSide,
+          transparent: true,
+          depthWrite: false,
+          blending: THREE.AdditiveBlending,
+        
+      });
+      //##################################################### pixel material #####################################################
+      const pixelMaterial = new THREE.ShaderMaterial({
+        vertexShader: `\
+            ${THREE.ShaderChunk.common}
+            ${THREE.ShaderChunk.logdepthbuf_pars_vertex}
+  
+            varying vec2 vUv;
+            varying vec3 vPos;
+            varying float vOpacity;
+            varying float vScales;
+            
+            attribute vec3 positions;
+            attribute float opacity;
+            attribute float scales;
+            
+        
+            void main() {
+                vUv=uv;
+                vPos = position;
+                vOpacity = opacity;
+                vScales = scales;
+                vec3 pos = position;
+                
+               
+                vec4 modelPosition = modelMatrix * vec4(position, 1.0);
+                vec4 viewPosition = viewMatrix * modelPosition;
+                vec4 projectionPosition = projectionMatrix * viewPosition;
+            
+                gl_Position = projectionPosition;
+                gl_PointSize = 35.0*scales;
+                vec4 mvPosition = modelViewMatrix * vec4( position, 1.0 );
+                bool isPerspective = ( projectionMatrix[ 2 ][ 3 ] == - 1.0 );
+                    if ( isPerspective ) gl_PointSize *= (1.0 / - viewPosition.z);
+                ${THREE.ShaderChunk.logdepthbuf_vertex}
+            }
+        `,
+        fragmentShader: `\
+            ${THREE.ShaderChunk.logdepthbuf_pars_fragment}
+            
+            varying vec2 vUv;
+            varying vec3 vPos;
+            varying float vOpacity;
+            varying float vScales;
+            void main() {
+                
+                gl_FragColor= vec4(0.0,vScales,0.3,1.0);
+                gl_FragColor.a *= vOpacity;
+                ${THREE.ShaderChunk.logdepthbuf_fragment}
+            }
+        `,
+        side: THREE.DoubleSide,
+        transparent: true,
+        depthWrite: false,
+        blending: THREE.AdditiveBlending,
+        
```

Comment:
There are a 3 properties we should add to our materials to make the shaders compile less, if the shader does not depend on lighting:

https://github.com/webaverse/app/blob/master/metaverse_modules/sonic-boom/index.js#L123

Reply:
add clipping fog lights
https://github.com/webaverse/app/pull/2873/commits/8e6aaeeb50c30349814493b408c403d6f8071805